### PR TITLE
feat: proxy pallet

### LINKED ISSUE #2460: Flexible multisig. Update multisig wallet discovery for getting flexible multisig
Update the multisig discovery procedure so that it will be able to differentiate the wallet type:
1. Regular multisig
2. Flexible multisig

Requirements [link](https://docs.google.com/document/d/1R1G5HoabNoUBweE0UhfRETIF9LdRcUQqGDRf28I_Umk/edit#heading=h.8wm85i4nogri)

### PR DESCRIPTION
**Pull request by johnthecat**
Partly closes #2460

### CODE REVIEW

**Review by sokolova-an — APPROVED** [review]

File: src/renderer/shared/pallet/referenda/storage.ts [hunk 1]
```diff
@@ -46,7 +47,7 @@ export const storage = {
     if (ids) {
       const schemaWithIds = pjsSchema
         .vec(pjsSchema.optional(referendaReferendumInfoConvictionVotingTally))
-        .transform(items => items.map((item, index) => ({ info: item, id: ids[index]! })));
+        .transform(items => zipWith(ids, items, (id, info) => ({ id, info })));
```
Comment (tuul-wq) [inline src/renderer/shared/pallet/referenda/storage.ts:50]: 👍

File: src/renderer/shared/pallet/proxy/storage.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    );
+
+    if (accounts && accounts.length > 0) {
+      const schema = pjsSchema.vec(recordSchema);
+
+      return substrateRpcPool
+        .call(() => getQuery(api, 'proxies').entries())
+        .then(schema.parse)
+        .then(result => zipWith(accounts, result, (account, value) => ({ account, value })));
+    } else {
```
Comment (tuul-wq) [inline src/renderer/shared/pallet/proxy/storage.ts]: No need for `else` when `if` contains `return`

File: src/renderer/shared/pallet/proxy/schema.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export const proxyProxyDefinition = pjsSchema.object({
+  delegate: pjsSchema.accountId,
+  delay: pjsSchema.u32,
+  proxyType: kitchensinkRuntimeProxyType,
+});
+
+export type ProxyAnnouncement = z.infer<typeof proxyAnnouncement>;
+export const proxyAnnouncement = pjsSchema.object({
+  real: pjsSchema.accountId,
+  callHash: pjsSchema.dataString,
```
Comment (Asmadek) [inline src/renderer/shared/pallet/proxy/schema.ts]: Can it be wrapped as something like hexString?

File: src/renderer/shared/pallet/proxy/storage.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    );
+
+    if (accounts && accounts.length > 0) {
+      const schema = pjsSchema.vec(recordSchema);
+
+      return substrateRpcPool
+        .call(() => getQuery(api, 'announcements').multi(accounts))
+        .then(schema.parse)
+        .then(result => zipWith(accounts, result, (account, value) => ({ account, value })));
+    } else {
```
Comment (Asmadek) [inline src/renderer/shared/pallet/proxy/storage.ts]: Is this else section really needed? we have return in if section. Same for others

**Review by Asmadek — APPROVED** [review]

**Pull request merged**